docs/basics.txt

```diff
@@ -635,23 +635,140 @@ bin/gremlin-server.sh ./conf/gremlin-server/http-gremlin-server.yaml
+```
+ graphs: {
+     credentialGraph: conf/janusgraph-cassandra.properties
+ }
+ ...
+ authentication: {
+   authenticator: org.janusgraph.graphdb.tinkerpop.gremlin.server.auth.JanusGraphSimpleAuthenticator,
+   authenticationHandler: org.apache.tinkerpop.gremlin.server.handler.HttpBasicAuthenticationHandler,
+   config: {
+     defaultUsername: user,
+     defaultPassword: password,
+     credentialsDb: conf/janusgraph-cassandra.properties
+   }
+ }
+```
+
+Verify that basic authentication is configured correctly. For example
 
 ```
-port: 8182
+curl -v -XPOST http://localhost:8182 -d '{"gremlin": "2+2"}'
+```
+
+should return a 401 if the authentication is configured correctly and
+
+```
+curl -v -XPOST http://localhost:8182 -d '{"gremlin": "2+2"}' -u user:password
+```
+should return a 200 and the result of 4 if authentication is configured correctly.
+
+===== HTTP HMAC authentication
+
+Another authentication method avaiablable, designed to amortize the number of rounds required to encrypt the password in basic auth, is HMAC authentication provided by JanusGraph. This is a token based authentication designed to be used over HTTP. You first acquire a token via the `/session` endpoint and then use that to authenticate.
+
+To enable HMAC token authentiation, the gremlin-server.yaml configuration should include the following properties
+
+```
+ graphs: {
+     credentialGraph: conf/janusgraph-cassandra.properties
+ }
+ ...
+ authentication: {
+   authenticator: org.janusgraph.graphdb.tinkerpop.gremlin.server.auth.HMACAuthenticator,
+   authenticationHandler: org.janusgraph.graphdb.tinkerpop.gremlin.server.handler.HttpHMACAuthenticationHandler,
+   config: {
+     defaultUsername: user,
+     defaultPassword: password,
+     hmacSecret: secret,
+     credentialsDb: conf/janusgraph-cassandra.properties
+   }
+ }
+```
+IMPORTANT: Note the hmacSecret here. This should be the same across all running JanusGraph servers if you want to be able to use the same HMAC token on each server.
+
+This creates a `/session` endpoint that provides a token that expires after an hour by default. This timeout for the token can be configured through the `tokenTimeout` configuration option in the `authentication.config` map. This value is a Long value and in milliseconds.
+
+You can obtain the token using curl by issuing a get request to the `/session` endpoint. For example
+
+```
+curl http://localhost:8182/session -XGET -u user:password
+
+{"token": "dXNlcjoxNTA5NTQ2NjI0NDUzOkhrclhYaGhRVG9KTnVSRXJ5U2VpdndhalJRcVBtWEpSMzh5WldqRTM4MW89"}
 ```
 
-In the future, the option to have one running instance of JanusGraph Server supporting multiple protocols may be possible.
+You can then use that token for authentication by using the "Authorization: Token" header. For example
 
+```
+curl -v http://localhost:8182/session -XPOST -d '{"gremlin": "1+1"}' -H "Authorization: Token dXNlcjoxNTA5NTQ2NjI0NDUzOkhrclhYaGhRVG9KTnVSRXJ5U2VpdndhalJRcVBtWEpSMzh5WldqRTM4MW89"
+```
```

Comment:
I was still able to authenticate with just username password with this configuration. Also it works with the full token but I found it also works when I change some of the trailing characters in the token. Is this expected?

Reply:
That doesn't sound right at all. That's a serious flaw and needs to be resolved.

Reply:
@robertdale You're right, that isn't correct. I assumed and well, we all know what happens when you assume. My apologies to you and @sjudeng . 

I was looking in to this, and this _isn't_ the time. This is the [base64 output padding](https://en.wikipedia.org/wiki/Base64#Output_padding).  

Note if you delete the last trailing `=` it doesn't work correctly. If you delete both, it does. You can also change the 5 last trailing characters (aside from the `==`), and things will work. This _isn't_ changing any of the significant parts of the HMAC token.

Reply:
@robertdale I just saw your other comment re: HMAC as well, I'll fix that today.